### ISSUE TITLE
NFS mounting issue

### ISSUE THREAD

**Issue by rabernat** [pangeo-data/pangeo-cloud-federation#84]
I am getting this in my event log as it tries to start my server

`2019-02-21 21:23:40+00:00 [Warning] MountVolume.SetUp failed for volume "ocean-staging-home-nfs" : mount failed: exit status 1 Mounting command: systemd-run Mounting arguments: --description=Kubernetes transient mount for /var/lib/kubelet/pods/a6af3b91-361e-11e9-94e8-42010a800086/volumes/kubernetes.io~nfs/ocean-staging-home-nfs --scope -- /home/kubernetes/containerized_mounter/mounter mount -t nfs <nil>:/ /var/lib/kubelet/pods/a6af3b91-361e-11e9-94e8-42010a800086/volumes/kubernetes.io~nfs/ocean-staging-home-nfs Output: Running scope as unit: run-r9a5ddc860e034ff9ac1e10447920c217.scope Mount failed: mount failed: exit status 32 Mounting command: chroot Mounting arguments: [/home/kubernetes/containerized_mounter/rootfs mount -t nfs <nil>:/ /var/lib/kubelet/pods/a6af3b91-361e-11e9-94e8-42010a800086/volumes/kubernetes.io~nfs/ocean-staging-home-nfs] Output: mount.nfs: Failed to resolve server <nil>: Name or service not known`

**Comment by yuvipanda** [comment]
It looks like somehow the NFS Server name is set to '<nil>' instead of the actual server name. That needs to be explicitly set - https://github.com/pangeo-data/pangeo-cloud-federation/blob/staging/deployments/dev/config/common.yaml#L62 sets it for dev.

This means you've to decide how you are going to do home directories. Options are:

1. Same home directory across all PANGEOs. This means you have one Google Cloud Filestore, same path, and every hub (in the same Google Cloud Project) shares it.
2. Different home directories per hub, but in same Google Cloud Filestore. You make a big cloud filestore, but use different paths (equivalent of https://github.com/pangeo-data/pangeo-cloud-federation/blob/staging/deployments/dev/config/common.yaml#L12) to give users one unique home directory per hub. This is probably the optimal choice now for most cases - you aren't super worried about raw performance, and you save on admin effort / cost. 
3. One google cloud filestore per hub. This gives you maximum performance, but costs most money - filestores have a minimum size, and one per hub makes for lower utilization.


I'd recommend going with (2) now, which would replicate the same values from https://github.com/pangeo-data/pangeo-cloud-federation/blob/staging/deployments/dev/config/common.yaml#L62 to ocean/config/common.yaml

**Comment by rabernat** [comment]
> 1. Same home directory across all PANGEOs. This means you have one Google Cloud Filestore, same path, and every hub (in the same Google Cloud Project) shares it.

^ This one

**Comment by jhamman** [comment]
Do we know how using the same home directory across all hubs will interact with the repo2docker components?

**Comment by rabernat** [comment]
I updated #87 to use the same home dir on both hubs. So we can test that question experimentally.

**Comment by rabernat** [comment]
This is still broken. I think #36 brought it back. Latest notebook log errors:

```
Server requested
2019-03-02 02:54:45+00:00 [Normal] Successfully assigned ocean-staging/jupyter-rabernat to gke-dev-pangeo-io-cl-n1-highmem-16-po-e8568f46-z8zw
2019-03-02 02:54:45+00:00 [Normal] MountVolume.SetUp succeeded for volume "daskkubernetes-token-xj4x7"
2019-03-02 02:54:45+00:00 [Warning] MountVolume.SetUp failed for volume "ocean-staging-home-nfs" : mount failed: exit status 1 Mounting command: systemd-run Mounting arguments: --description=Kubernetes transient mount for /var/lib/kubelet/pods/8930ae4f-3c96-11e9-8db7-42010a8000ab/volumes/kubernetes.io~nfs/ocean-staging-home-nfs --scope -- /home/kubernetes/containerized_mounter/mounter mount -t nfs <nil>:/ /var/lib/kubelet/pods/8930ae4f-3c96-11e9-8db7-42010a8000ab/volumes/kubernetes.io~nfs/ocean-staging-home-nfs Output: Running scope as unit: run-r69e275890ced4868a42150b1ada93dde.scope Mount failed: mount failed: exit status 32 Mounting command: chroot Mounting arguments: [/home/kubernetes/containerized_mounter/rootfs mount -t nfs <nil>:/ /var/lib/kubelet/pods/8930ae4f-3c96-11e9-8db7-42010a8000ab/volumes/kubernetes.io~nfs/ocean-staging-home-nfs] Output: mount.nfs: Failed to resolve server <nil>: Name or service not known
2019-03-02 02:54:46+00:00 [Warning] MountVolume.SetUp failed for volume "ocean-staging-home-nfs" : mount failed: exit status 1 Mounting command: systemd-run Mounting arguments: --description=Kubernetes transient mount for /var/lib/kubelet/pods/8930ae4f-3c96-11e9-8db7-42010a8000ab/volumes/kubernetes.io~nfs/ocean-staging-home-nfs --scope -- /home/kubernetes/containerized_mounter/mounter mount -t nfs <nil>:/ /var/lib/kubelet/pods/8930ae4f-3c96-11e9-8db7-42010a8000ab/volumes/kubernetes.io~nfs/ocean-staging-home-nfs Output: Running scope as unit: run-r84594d07792544868f0705c929b778b3.scope Mount failed: mount failed: exit status 32 Mounting command: chroot Mounting arguments: [/home/kubernetes/containerized_mounter/rootfs mount -t nfs <nil>:/ /var/lib/kubelet/pods/8930ae4f-3c96-11e9-8db7-42010a8000ab/volumes/kubernetes.io~nfs/ocean-staging-home-nfs] Output: mount.nfs: Failed to resolve server <nil>: Name or service not known
...
...
```

**Comment by rabernat** [comment]
#94 didn't work. hubploy failed with this interesting error

```
#!/bin/bash -eo pipefail
hubploy deploy ocean pangeo-deploy ${CIRCLE_BRANCH}
Activated service account credentials for: [hubploy-circleci@pangeo-181919.iam.gserviceaccount.com]
WARNING: Accessing a Kubernetes Engine cluster requires the kubernetes commandline
client [kubectl]. To install, run
  $ gcloud components install kubectl

Fetching cluster endpoint and auth data.
kubeconfig entry generated for dev-pangeo-io-cluster.
Hang tight while we grab the latest from your chart repositories...
...Unable to get an update from the "local" chart repository (http://127.0.0.1:8879/charts):
	Get http://127.0.0.1:8879/charts/index.yaml: dial tcp 127.0.0.1:8879: connect: connection refused
...Successfully got an update from the "jupyterhub" chart repository
...Successfully got an update from the "stable" chart repository
Update Complete. ⎈Happy Helming!⎈
Saving 1 charts
Downloading jupyterhub from repo https://jupyterhub.github.io/helm-chart/
Deleting outdated charts
Error: UPGRADE FAILED: PersistentVolume "ocean-staging-home-nfs" is invalid: spec.persistentvolumesource: Forbidden: is immutable after creation
Traceback (most recent call last):
  File "/root/repo/venv/bin/hubploy", line 11, in <module>
    load_entry_point('hubploy==0.1.0', 'console_scripts', 'hubploy')()
  File "/root/repo/venv/lib/python3.6/site-packages/hubploy/__main__.py", line 57, in main
    helm.deploy(args.deployment, args.chart, args.environment, args.namespace, args.set, args.version)
  File "/root/repo/venv/lib/python3.6/site-packages/hubploy/helm.py", line 130, in deploy
    version
  File "/root/repo/venv/lib/python3.6/site-packages/hubploy/helm.py", line 60, in helm_upgrade
    subprocess.check_call(cmd)
  File "/usr/lib/python3.6/subprocess.py", line 291, in check_call
    raise CalledProcessError(retcode, cmd)
subprocess.CalledProcessError: Command '['helm', 'upgrade', '--wait', '--install', '--namespace', 'ocean-staging', 'ocean-staging', 'pangeo-deploy', '-f', 'deployments/ocean/config/common.yaml', '-f', 'deployments/ocean/config/staging.yaml', '-f', 'deployments/ocean/secrets/staging.yaml', '--set', 'jupyterhub.singleuser.image.tag=64ce6b3', '--set', 'jupyterhub.singleuser.image.name=us.gcr.io/pangeo-181919/ocean-pangeo-io-notebook']' returned non-zero exit status 1.
Exited with code 1
```

**Comment by rabernat** [comment]
I just tried manually deleting the `ocean-staging-home-nfs` volume claim. It disappeared, but then when I reran the hubploy deploy workflow via circle ci, it was re-created with the same error.

**Comment by rabernat** [comment]
fixed by #94
